src/commands/check.rs

```diff
@@ -182,17 +257,17 @@ pub async fn check(args: CheckArgs) -> anyhow::Result<()> {
 
     if error_count > 0 {
         bail!(
-            "aborting due to previous {error_count} error{s}",
+            "failing due to {error_count} error{s}",
             s = if error_count == 1 { "" } else { "s" }
         );
     } else if args.common.deny_warnings && warning_count > 0 {
         bail!(
-            "aborting due to previous {warning_count} warning{s} (`--deny-warnings` was specified)",
+            "failing due to {warning_count} warning{s} (`--deny-warnings` was specified)",
             s = if warning_count == 1 { "" } else { "s" }
         );
     } else if args.common.deny_notes && note_count > 0 {
         bail!(
-            "aborting due to previous {note_count} note{s} (`--deny-notes` was specified)",
+            "failing due to {note_count} note{s} (`--deny-notes` was specified)",
```

Comment:
I'm not sure about this language change. I'm not sure the new proposed language makes it clear that `sprocket` is stopping because it has encountered too many errors/notes/warning.

Reply:
I think that this language is an improvement, but maybe it can be improved further. I don't like the `aborting` language because in my mind it implies that `sprocket` was going to do something else but it's quitting early. This really just controls whether it's a non-zero exit code or not.

Reply:
Can you clarify the behavior here? `sprocket` is computing the full set of diagnostics, reporting all of them, and then setting the exit code if any of the requested subtypes has content?

Reply:
You are correct. The `bail!` macro exits the program with a non-zero exit code (IDK what code specifically) and this block of checks is the last thing `sprocket` is doing in `check/lint` mode. So there is no further computation that _might_ occur if there aren't problems detected. This is the end no matter what.

Reply:
OK, I initially thought this was related to limiting the number of diagnostics emitted to the first `N`. I know that was discussed, but can't recall what the decision was.

Now that I understand it, I agree that the language is an improvement.

Reply:
We did discuss that over slack, but it was decided that limiting to the first `N` would likely lead to a bad UX. So we always report the full set of diagnostics, with `--local-only` and `--single-document` controlling the scope of what's reported (but everything is always computed regardless)

Reply:
Thanks for the refresher, the after-effects of the holiday break are quite apparent today.